localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/parse.py

```diff
@@ -99,3 +111,44 @@ def _get_single_and_multi_values_from_headers(
             multi_values[key] = headers.getlist(key)
 
         return single_values, multi_values
+
+    @staticmethod
+    def create_context_variables(context: RestApiInvocationContext) -> ContextVariables:
+        invocation_request: InvocationRequest = context.invocation_request
+        domain_name = invocation_request["raw_headers"].get("Host", "")
+        domain_prefix = domain_name.split(".")[0]
+        now = datetime.datetime.now()
+
+        # TODO: verify which values needs to explicitly have None set
+        context_variables = ContextVariables(
+            accountId=context.account_id,
+            apiId=context.api_id,
+            deploymentId=context.deployment_id,
+            domainName=domain_name,
+            domainPrefix=domain_prefix,
+            extendedRequestId=short_uid(),  # TODO: use snapshot tests to verify format
+            httpMethod=invocation_request["http_method"],
+            path=invocation_request[
+                "path"
+            ],  # TODO: check if we need the raw path? with forward slashes
+            protocol="HTTP/1.1",
+            requestId=short_uid(),  # TODO: use snapshot tests to verify format
+            requestTime=timestamp(time=now, format=REQUEST_TIME_DATE_FORMAT),
+            requestTimeEpoch=int(now.timestamp() * 1000),
+            stage=context.stage,
+        )
+        return context_variables
```

Comment:
Should we log this dict immediately?

Reply:
Done! Good catch! We can revisit the overall logging later on when we have a clear picture, to make it a consistent experience, but for now simply logging it is really good. I believe we will also realize later on some values needs default values, maybe `$context.identity` for example?